src/org/mirah/jvm/mirrors/generics/xx_type_invoker.mirah

```diff
@@ -100,16 +100,20 @@ class TypeInvoker < BaseSignatureReader
   end
 
   def self.invoke(context:Context, type:MirrorType, args:List,
-                  outerTypeArgs:Map):MirrorType
+                  outerTypeArgs:Map, processed_signatures:Map):MirrorType
     dtype = DeclaredMirrorType(type)
     if dtype.signature.nil? || args.any? {|a| a.nil?}
       type
     else
-      invoker = TypeInvoker.new(context, outerTypeArgs, args)
-      invoker.read(dtype.signature)
+      invoker = TypeInvoker processed_signatures[dtype.signature]
```

Comment:
Kinda looks more Java-esque that way :) It seems more visually obvious that it's a cast and not a method call.  In older versions of Mirah I had problems with casts being mistaken for method calls and I wondered why the Java style cast was abandoned but I don't have problems with that anymore.